### PR TITLE
[RSDK-4960] Publish workflow

### PR DESCRIPTION
**Pull request by npmenard**
Automatically publish a esp32-serrver binary when creating a tag. Treat rc release as pre-release

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        fi
+    - name: Publish release candidate
+      if: steps.check-tag.outputs.match == 'true'
+      uses: marvinpinto/action-automatic-releases@latest
+      with:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+        files: |
+          micro-rdk-esp32-server.bin
+          sha256sums.txt
+        prerelease: true
```
Comment (acmorrow) [inline .github/workflows/publish.yml:42]: Could you dedup this with something like `prerelease: ${{needs.check-tag.outputs.output1}}` per https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs?

I think you would need to make "Publish" its own job though. So something like:

```
jobs:
  tests: ...
  build-release:
      outputs:
        prerelease: ${{ steps.check-tag.outputs.prerelease }}
    steps:
     - name: Checkout...
     - name: Build ...
     - name: Check release type
        id: check-tag
        run: |
          if echo ${{ github.event.ref }} | grep -Eq '^refs/tags/v.*rc[0-9]{1}$'; then
                echo "prerelease=true" >> $GITHUB_OUTPUT
          else
                echo "prerelease=false" >> $GITHUB_OUTPUT
          fi
  publish-release:
    needs: build-release
    steps:
    - publish:
      uses: marvinpinto/action-automatic-releases@latest
      with:
        repo_token: ${{ secrets.GITHUB_TOKEN }}
        files: |
          micro-rdk-esp32-server.bin
          sha256sums.txt
        prerelease: ${{needs.build-release.outputs.prerelease}}
```

Something like that.

**Review by acmorrow — APPROVED** [review]
LGTM with an optional suggestion to remove the duplication around `prerelease`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -37,16 +42,16 @@ jobs:
       with:
         filters: |
           src-esp32:
-            - 'src/esp32/**'
+            - 'src/esp3233/**'
```
Comment (acmorrow) [inline .github/workflows/test.yml:45]: Was this maybe left over from some debugging?

Comment (npmenard) [reply .github/workflows/test.yml:45]: yes I'll remove them before merge

**Pull request merged**